lib/openssl/ssl.rb

```diff
@@ -314,7 +314,7 @@ def verify_hostname(hostname, san) # :nodoc:
       san_parts = san.downcase.split(".")
 
       # TODO: this behavior should probably be more strict
-      return san == hostname if san_parts.size < 2
+      return san.downcase == hostname.downcase if san_parts.size < 2
```

Comment:
ssl.rb is mostly copied from openssl gem: https://github.com/ruby/openssl/blob/master/lib/openssl/ssl.rb#L317

you maybe want to post this upstream first, ideally with an actual test-case.